### PR TITLE
Add array and map types

### PR DESCRIPTION
**Pull request by mateuszmlc**
Allows the user to define fields of type array `string[]`, `number[]` and map `map<string | number, string | number>`

### CODE REVIEW

**Comment by calummoore** [comment]
I was thinking the `map` would be recursive. So map type would be: `type MapValue = string | number | number[] | string[], map<string|number, MapValue>` and `map<string|number, MapValue>`. 

Would that add a lot of complexity?

**Comment by mateuszmlc** [comment]
It shouldn't, I'll try to add that

**Comment by mateuszmlc** [comment]
Implemented 👍

File: src/lib.rs [hunk 1]
```diff
@@ -537,9 +537,73 @@ function x() {
         eprintln!("{}", collection.as_ref().unwrap_err().message);
         assert_eq!(
             collection.unwrap_err().message,
-            r#"Error found at line 3, column 22: Unrecognized token "object". Expected one of: "number", "string"
+            r#"Error found at line 3, column 22: Unrecognized token "object". Expected one of: "map", "number", "string"
```
Comment (calummoore) [inline src/lib.rs:540]: Should we not include array in this?

**Review by calummoore — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (mateuszmlc) [reply src/lib.rs:540]: Array is parsed as 3 tokens, ("number" OR "string") + "[" + "]", the message is generated by Lalrpop and it only shows the next expected token

**Review by calummoore — APPROVED** [review]

**Pull request merged**